### PR TITLE
Add testing of `testeth --filltests` to Travis

### PR DESCRIPTION
**Pull request by jwasinger**
Related issue: https://github.com/ethereum/cpp-ethereum/issues/4369
Depends on: https://github.com/ethereum/tests/pull/268, https://github.com/ethereum/tests/pull/275

### CODE REVIEW

**Comment by pirapira** [comment]
Please rebase.

**Comment by jwasinger** [comment]
@pirapira can you update the tests submodule to point to the latest commit on https://github.com/ethereum/tests?

**Comment by pirapira** [comment]
@jwasinger I would file a pull-request for doing that.  You can also do that.

Now I remember that this PR depends on https://github.com/ethereum/tests/pull/268 .  After that, maybe this PR does not need any commit that modifies `test/jsontests` submodule.

**Comment by jwasinger** [comment]
@pirapira Agreed.  PR for rebase: https://github.com/ethereum/cpp-ethereum/pull/4399

**Comment by winsvega** [comment]
why there are so many changes in tests submodule?

**Comment by jwasinger** [comment]
@winsvega Not sure if there is an issue?  The checksum of the submodule is up to date with tests/develop.

**Comment by pirapira** [comment]
@jwasinger can you try to remove the mistaken commits from this PR?  If you don't know `git rebase -i`, read about it.

**Comment by pirapira** [comment]
@jwasinger please rebase.  Adding new commits will not solve the conflict.  Read about `git rebase` if you don't know about it.

**Comment by jwasinger** [comment]
@pirapira I see the issue.  Submodule change was unexpectedly introduced on the first commit.  Removing it.  I'm still learning how to use Git.

**Comment by pirapira** [comment]
Git submodules are complicated enough.

**Comment by jwasinger** [comment]
Okay.  Changes are looking good to me.  In order to make sure the new Travis tests for `testeth --filltests` are working need to merge https://github.com/ethereum/tests/pull/275 and then https://github.com/ethereum/tests/pull/268

**Comment by pirapira** [comment]
I merged the dependencies in `ethereum/tests`.

**Comment by pirapira** [comment]
Getting close!

**Comment by jwasinger** [comment]
I rebased.  Travis should (hopefully) pass now that the tests submodule is up to date.

**Comment by jwasinger** [comment]
YAY! Linting of tests produced by `testeth` is working in Travis.  Cleaning up the commit history and then, this should be ready for merging.

File: scripts/tests.sh [hunk 1]
```diff
@@ -30,12 +30,21 @@ TESTS=$1
...
 BUILD_ROOT=$(pwd)
+pwd
 
 if [[ "$TESTS" == "On" ]]; then
 
     # Run the tests for the Interpreter
     $BUILD_ROOT/test/testeth -- --testpath $BUILD_ROOT/../test/jsontests
 
+    # Fill some state tests and make sure the result passes linting
+    echo "Running testeth filler tests..." && \
```
Comment (pirapira) [inline scripts/tests.sh]: There is `set -e` on line 27, so `&&`-chaining is not necessary.

**Comment by pirapira** [comment]
I'm pulling in @chfast .  He might know a better way to install the LLL compiler and NodeJS.

**Comment by chfast** [comment]
Does solidity package provides lllc, @chriseth?
For NodeJS, the best would be to request it from Travis CI. You can check Emscripten build for solidity.

**Comment by axic** [comment]
@chfast nothing includes lllc, I have this issue for it: https://github.com/ethereum/solidity/issues/2785

**Comment by pirapira** [comment]
@jwasinger are you going to change the way nodejs is installed?  See @chfast's comment above.

**Comment by jwasinger** [comment]
@pirapira Yes.  Currently in the middle of a move to a new house.  I will wrap this PR up today.

**Comment by jwasinger** [comment]
The time difference also makes it difficult.  I only just saw @chfast 's comment 6 hours after he posted it due to just waking up (8am PST).

**Comment by pirapira** [comment]
@jwasinger No worries. I'm aware of the time difference.  Actually, that's the reason I asked ahead.  If we shared a timezone, I would have waited into the afternoon before asking about your next move.  My psychological time warped.

**Comment by jwasinger** [comment]
ah i see :)

**Comment by jwasinger** [comment]
I think this is ready to merge @pirapira

**Comment by winsvega** [comment]
keep in mind that original solidity version might not support some of the new opcodes

**Comment by chfast** [comment]
How the fuck it happened that now we are building full solidity from source during cpp-ethereum build on Travis? Remove this now!

**Comment by jwasinger** [comment]
Working on the PR now

**Comment by jwasinger** [comment]
@chfast

**Comment by chfast** [comment]
@jwasinger sync with @pirapira.

We've noticed that lllc can be downloaded from https://github.com/ethereum/solidity/releases.
And you can reuse this script to do this: https://github.com/ethereum/cpp-ethereum/blob/develop/scripts/install_cmake.sh.

**Pull request merged**